heareval/task_embeddings.py

```diff
@@ -0,0 +1,142 @@
+
+
+class CSVDataset(Dataset):
+    """
+    Read in a CSV file, and return data rows as [column 1, (column 2, ...)]
+    """
+
+    def __init__(self, csv_file):
+        # Our CSV files don't have headers, so we can't use
+        # pd.read_csv
+        # I'm on the fence whether we want CSV headers or not,
+        # since the format is standardized.
+        with open(csv_file) as f:
+            csvreader = csv.reader(f)
+            self.rows = [row for row in csvreader]
+        ncol = len(self.rows[0])
+        # Make sure all rows have the same number of column,
+        # and rewrite as [col1, (col2, ...)]
+        for idx, row in enumerate(self.rows):
+            assert len(row) == ncol
+            self.rows[idx] = [row[0], row[1:]]
+
+    def __len__(self):
+        return len(self.rows)
+
+    def __getitem__(self, idx):
+        return self.rows[idx]
+
+
+# These is a high-level wrappers on the basic API that we might
+# consider sharing among all participants.
+
+
+def get_audio_embedding_numpy(
+    audio_numpy: np.ndarray,
+    model: Any,
+    frame_rate: float,
+) -> Tuple[Dict[int, np.ndarray], np.ndarray]:
+    # Is there a way to avoid this .float() cast? Or do it on the
+    # numpy, not after the tensor creation?
+    embedding_dict, timestamps = EMBED.get_audio_embedding(  # type: ignore
+        torch.tensor(audio_numpy, device=device).float(),
+        model=model,
+        frame_rate=frame_rate,
+    )
+    for key in embedding_dict.keys():
+        embedding_dict[key] = embedding_dict[key].detach().cpu().numpy()
+    timestamps = timestamps.detach().cpu().numpy()
+    return embedding_dict, timestamps
+
+
+if __name__ == "__main__":
+    model = EMBED.load_model(EMBEDDING_MODEL_PATH, device=device)  # type: ignore
+
+    # TODO: Would be good to include the version here
+    # https://github.com/neuralaudio/hear2021-eval-kit/issues/37
+    embeddir = os.path.join("embeddings", EMBED.__name__)  # type: ignore
+    embedsr = EMBED.input_sample_rate()  # type: ignore
+
+    for task in glob.glob("tasks/*"):
+        # TODO: We should be reading the metadata that describes
+        # the frame_rate.
+        # https://github.com/neuralaudio/hear2021-eval-kit/issues/53
+        frame_rate = 10
+
+        # TODO: Include "val" ?
+        for split in ["train", "test"]:
+            # TODO: We might consider skipping files that already
+            # have embeddings on disk, for speed
+            dataloader = DataLoader(
+                CSVDataset(os.path.join(task, f"{split}.csv")),
+                batch_size=64,
+                shuffle=True,
+            )
+            outdir = os.path.join(embeddir, task, split)
+            if not os.path.exists(outdir):
+                os.makedirs(outdir)
+            for batch in tqdm(dataloader):
+                files, labels = batch
+                audios = []
+                for f in files:
+                    x, sr = sf.read(os.path.join(task, str(embedsr), split, f))
```

Comment:
`sf.read(..., dtype=np.flaot32)` then you won't have to cast to float in the tensor above

Reply:
Nice, will fix